tests/ImageSharp.Tests/Memory/Allocators/UniformUnmanagedMemoryPoolTests.Trim.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using Microsoft.DotNet.RemoteExecutor;
+using SixLabors.ImageSharp.Memory.Internals;
+using Xunit;
+
+namespace SixLabors.ImageSharp.Tests.Memory.Allocators
+{
+    public partial class UniformUnmanagedMemoryPoolTests
+    {
+        [Collection(nameof(NonParallelTests))]
+        public class Trim
+        {
+            [CollectionDefinition(nameof(NonParallelTests), DisableParallelization = true)]
+            public class NonParallelTests
+            {
+            }
+
+            [Fact]
+            public void TrimPeriodElapsed_TrimsHalfOfUnusedArrays()
+            {
+                RemoteExecutor.Invoke(RunTest).Dispose();
+                static void RunTest()
+                {
+                    var trimSettings = new UniformUnmanagedMemoryPool.TrimSettings { TrimPeriodMilliseconds = 5_000 };
+                    var pool = new UniformUnmanagedMemoryPool(128, 256, trimSettings);
+
+                    UnmanagedMemoryHandle[] a = pool.Rent(64);
+                    UnmanagedMemoryHandle[] b = pool.Rent(64);
+                    pool.Return(a);
+                    Assert.Equal(128, UnmanagedMemoryHandle.TotalOutstandingHandles);
+                    Thread.Sleep(15_000);
+
+                    // We expect at least 2 Trim actions, first trim 32, then 16 arrays.
+                    // 128 - 32 - 16 = 80
+                    Assert.True(
+                        UnmanagedMemoryHandle.TotalOutstandingHandles <= 80,
+                        $"UnmanagedMemoryHandle.TotalOutstandingHandles={UnmanagedMemoryHandle.TotalOutstandingHandles} > 80");
+                    pool.Return(b);
+                }
+            }
+
+            public static readonly bool IsNotMacOs = !TestEnvironment.IsOSX;
+
+            // TODO: Investigate failures on MacOS. All handles are released after GC.
```

Comment:
Maybe worth raising an issue to track.

Reply:
Yeah will do after the merge, when it officially becomes the product code.